filebeat/input/filestream/internal/input-logfile/harvester.go

```diff
@@ -189,10 +189,17 @@ func startHarvester(
 
 		harvesterCtx, cancelHarvester, err := hg.readers.newContext(srcID, canceler)
 		if err != nil {
-			// The returned may or not be collected by the caller, thus logging
-			// it here is important.
-			ctx.Logger.Errorf("error while adding new reader to the bookkeeper %v", err)
-			return fmt.Errorf("error while adding new reader to the bookkeeper %w", err)
+			// The only possible returned error is ErrHarvesterAlreadyRunning, which is a normal
+			// behaviour of the Filestream input, it's not really an error, it's just an situation.
+			// If the harvester is already running we don't need to start a new one.
+			// At the moment of writing even the returned error is ignored. So the
+			// only real effect of the next line is to not start a second harvester.
+			//
+			// Currently the only places this error is checked is on task.Group and the
+			// only thing it does is to log the error. So to avoid unnecessary errors,
+			// we just return nil.
+			//nolint: nilerr // Read the comment above
+			return nil
```

Comment:
I agree that it's not an error but we still should log it at least at the debug level.

Reply:
This path can be triggered by events watched by the file watcher, which can be a lot, even o debug level I believe it's too verbose.

I let it running on my machine for a while, in 2h there were about 2000 of those messages for a single active file (the Agent's log file). They're duplicated because the `task.Group` also logs it, so about 500 messages per hour per file if it's logged on debug mode. I still believe it's too much.

What do you think?

Reply:
I think it's okay in the current form.